src/Directory.Build.targets

```
@@ -16,9 +16,52 @@
           AfterTargets="Pack"
           Condition="'$(IsPackable)' == 'true' and '$(BlobGroupPrefix)' != ''">
     <PropertyGroup>
-      <_BlobGroupVersionMajor>$(PackageVersion.Split('.')[0])</_BlobGroupVersionMajor>
-      <_BlobGroupVersionMinor>$(PackageVersion.Split('.')[1])</_BlobGroupVersionMinor>
-      <_BlobGroupName>$(BlobGroupPrefix)$(_BlobGroupVersionMajor).$(_BlobGroupVersionMinor)</_BlobGroupName>
+      <!--
+        These properties take a package version and transform it into a blob group name so that
+        all builds from the same product and release version are grouped together. This code has
+        to consider when the version is a release version (e.g. 5.0.0) or has a prerelease label
+        (e.g. 5.0.0-preview.1). The former is transformed into '$(BlobGroupPrefix)5.0/release' whereas
+        the latter is transformed into '$(BlobGroupPrefix)5.0/preview.1'. It also accounts for the
+        BlobGroupBuildQuality defined in Version.props, which determines if the prerelease information
+        should be used in the final blob group name.
+        -->
+      <_PreReleaseSeperatorIndex>$(PackageVersion.IndexOf('-'))</_PreReleaseSeperatorIndex>
+      
+      <!-- Prerelease: '5.0.0-preview.1' -> '5.0.0' and 'preview.1' -->
+      <_BlobGroupVersion Condition="'$(_PreReleaseSeperatorIndex)' != '-1'">$(PackageVersion.Substring(0, $(_PreReleaseSeperatorIndex)))</_BlobGroupVersion>
+      <_BlobGroupPreRelease Condition="'$(_PreReleaseSeperatorIndex)' != '-1'">$(PackageVersion.Substring($([MSBuild]::Add($(_PreReleaseSeperatorIndex), 1))))</_BlobGroupPreRelease>
+      
+      <!-- Release: take the package version as-is. -->
+      <_BlobGroupVersion Condition="'$(_PreReleaseSeperatorIndex)' == '-1'">$(PackageVersion)</_BlobGroupVersion>
+
+      <!--
+        Prerelease may contain additional fields; only want the label and first number
+        (e.g. 'preview.1.12345' -> 'preview' and '1').
+        -->
+      <_BlobGroupPreReleaseLabel Condition="'$(_BlobGroupPreRelease)' != ''">$(_BlobGroupPreRelease.Split('.')[0])</_BlobGroupPreReleaseLabel>
+      <_BlobGroupPreReleaseIteration Condition="'$(_BlobGroupPreRelease)' != ''">$(_BlobGroupPreRelease.Split('.')[1])</_BlobGroupPreReleaseIteration>
+
+      <!--
+        Calculate release name based on version parts and blob group build quality.
+        Daily: use 'daily'
+        Prerelease: 'preview' and '1' -> 'preview.1'
+        Release: use 'release'
+        -->
+      <_BlobGroupReleaseName  Condition="'$(BlobGroupBuildQuality)' == 'daily'">daily</_BlobGroupReleaseName>
+      <_BlobGroupReleaseName  Condition="'$(BlobGroupBuildQuality)' == 'prerelease'">$(_BlobGroupPreReleaseLabel).$(_BlobGroupPreReleaseIteration)</_BlobGroupReleaseName>
+      <_BlobGroupReleaseName  Condition="'$(BlobGroupBuildQuality)' == 'release'">release</_BlobGroupReleaseName>
```

Maybe error out if `_BlobGroupReleaseName` is empty after this point. 
```suggestion
      <_BlobGroupReleaseName  Condition="'$(BlobGroupBuildQuality)' == 'release'">release</_BlobGroupReleaseName>
      <Error
            Text="The BlobGroupBuildQuality property must be set to one of: daily, release, prerelease."
            Condition="'$(_BlobGroupReleaseName)' == ''" />
```

---

I'll make this change, but it'll have to be outside of the PropertyGroup.

---

I created an item group of the valid names and validated against that as well as validating the blob group parts are not empty.